File: 2491-divide-players-into-teams-of-equal-skill/2491-divide-players-into-teams-of-equal-skill.java
class Solution {
    public long dividePlayers(int[] skill) {
        long ans=0;
        int i=0,j=skill.length-1;
        Arrays.sort(skill);
        int sum=skill[i]+skill[j];
        ans=skill[i]*skill[j];
        i++;j--;
        while(i<j){
            if(skill[i]+skill[j]!=sum){
                return -1;
            }
            ans+=(skill[i]*skill[j]);
            i++;
            j--;
            
        }
        return ans;
    }
}

File: 2491-divide-players-into-teams-of-equal-skill/2491-divide-players-into-teams-of-equal-skill.cs
public class Solution {
    public long DividePlayers(int[] skill) {
        Array.Sort(skill);
        int i=0,j=skill.Count()-1;
        long ans=0;
        while(i<j){
            if(skill[i]+skill[j]!=skill[0]+skill[skill.Count()-1])
                return -1;
            ans=ans+skill[i]*skill[j];
            i++;
            j--;
        }
        return ans;
    }
}